spec/lib/services/api/metric_rollups_service_spec.rb

```diff
@@ -0,0 +1,19 @@
+describe Api::MetricRollupsService do
+  before do
+    allow(Settings.api).to receive(:metrics_default_limit).and_return(1000)
+  end
+
+  it 'validates that all of the parameters are present' do
```

Comment:
Since all these tests are about validation, I'm wondering it that is suggesting the service's single responsibility should be validation, and that the rest could be pulled out either into the caller, or another service?